### PR TITLE
Handle hidden dot files in iOS framework bundles

### LINKED ISSUE #99229: 'Flutter/Flutter.h' file not found
## Steps to Reproduce
Apple M1 Mac
Cocoapods installed via brew
Using either Visual Studio Code or Android Studio
Apple M1 Mac
Use a dependency such as path_provider 
flutter get pub
set device simulator to iPhone (13 pro)
flutter run

**Expected results:** 
App to Launch successfully

**Actual results:** 
Error
Lexical or Preprocessor Issue (Xcode): 'Flutter[/Flutter.h]()' file not found
[/Volumes/T7/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider_ios-2.0.8/ios/Classes/FLTPathProviderPlugin.h:4:8]()

<details>
  <summary>Logs</summary>

Attached is the flutter run -verbose
[flutter-run-verbose-header file not found issue.txt](https://github.com/flutter/flutter/files/8148511/flutter-run-verbose-header.file.not.found.issue.txt)

Screenshot of iOS/build folder
Screenshot 2022-02-27 at 10 46 59

flutter doctor -v
[✓] Flutter (Channel stable, 2.10.2, on macOS 12.2.1 21D62 darwin-arm, locale en-GB)
    • Flutter version 2.10.2 at /Volumes/T7/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 097d3313d8 (8 days ago), 2022-02-18 19:33:08 -0600
    • Engine revision a83ed0e5e3
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/blumf/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /Library/Java/JavaVirtualMachines/jdk-17.0.2.jdk/Contents/Home/bin/java
    • Java version Java(TM) SE Runtime Environment (build 17.0.2+8-LTS-86)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] VS Code (version 1.64.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.34.0

[✓] Connected device (2 available)
    • iPhone 13 Pro (mobile) • E792BB5C-84A9-4F3E-A18F-D1CBDFA19580 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-2 (simulator)
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 98.0.4758.109

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

**Additional Notes**
I did some analysis on the verbose output and I noticed something odd in the Build Settings:

FRAMEWORK_SEARCH_PATHS =  "/Volumes/T7/initiative_tracker2/build/ios/Debug-iphoneos/path_provider_ios"
HEADER_SEARCH_PATHS =  "/Volumes/T7/initiative_tracker2/build/ios/Debug-iphoneos/path_provider_ios/path_provider_ios.framework/Headers"

The 'Debug-iphoneos' folder does not exist in the build/iOS folder, but there is a 'Debug-iphonesimulator'
Please see attached screenshot.

It could be that the incorrect header/framework search paths are being sent to Xcode and it is unable to find the dependency package headers. 

</details>

### PR DESCRIPTION
**Pull request by jmagman**
Fixes https://github.com/flutter/flutter/issues/99229

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/devicelab/bin/tasks/module_test_ios.dart [hunk 1]
```diff
@@ -79,6 +79,26 @@ Future<void> main() async {
         );
       }
 
+      section('Build ephemeral host app when SDK is on external disk');
+
+      // Don't clean, pretend the SDK was on an external drive with stray "._" files in the xcframework
```
Comment (christopherfujino) [inline dev/devicelab/bin/tasks/module_test_ios.dart:84]: is this section dependent on running right after the previous section?

**Review by christopherfujino — APPROVED** [review]
LGTM

File: dev/devicelab/bin/tasks/module_test_ios.dart (hunk 1 above)
Comment (jmagman) [reply dev/devicelab/bin/tasks/module_test_ios.dart:84]: No, I don't need the "Don't clean" part of this comment it doesn't make sense.
```suggestion
      // Pretend the SDK was on an external drive with stray "._" files in the xcframework.
```

File: dev/devicelab/bin/tasks/module_test_ios.dart [hunk 2]
```diff
@@ -79,6 +79,26 @@ Future<void> main() async {
         );
       }
 
+      section('Build ephemeral host app when SDK is on external disk');
+
+      // Pretend the SDK was on an external drive with stray "._" files in the xcframework.
```
Comment (jmagman) [inline dev/devicelab/bin/tasks/module_test_ios.dart:84]:
```suggestion
      // Pretend the SDK was on an external drive with stray "._" files in the xcframework
```

**Comment by casyalex** [comment]
HI, When wil this fix be release,d i checked it is not include at 3.0.5 tag. But it has been two months since it was merged...

**Comment by jmagman** [comment]
@casyalex It became available in 2.13.0-0.1.pre, so the fix is also in 3.0.5.  Just confirmed:

```
$ git co 3.0.5
HEAD is now at f1875d570e [flutter_releases] Flutter stable 3.0.5 Framework Cherrypicks (#107565)
$ grep -riI "Hidden file" packages/flutter_tools/bin/podhelper.rb
packages/flutter_tools/bin/podhelper.rb:      continue if xcframework_file.start_with?(".") # Hidden file, possibly on external disk.
```

If you're still seeing a problem, please file a new issue and completely fill out the template. Thanks!

**Comment by casyalex** [comment]
https://github.com/flutter/flutter/blob/f1875d570e39de09040c8f79aa13cc56baab8db1/packages/flutter_tools/bin/podhelper.rb#L63 #

@jmagman Might got override or some how, please check above link?

I alos check my local 3.0.5 source code, the **continue** still there..

**Comment by jmagman** [comment]
@casyalex https://github.com/flutter/flutter/pull/104296 is in 3.1.0-9.0.pre.  Please see https://github.com/flutter/flutter/wiki/Where's-my-Commit%3F#finding-the-versions-that-contain-framework-commit-x

**Pull request merged**